src/api/java/FuncDecl.java

```diff
@@ -29,35 +29,22 @@
     /**
      * Object comparison.
      **/
+    @Override
     public boolean equals(Object o)
     {
-        FuncDecl casted = null;
-
-        try {
-            casted = FuncDecl.class.cast(o);
-        } catch (ClassCastException e) {
-            return false;
-        }
-
-	return  
-	    (this == casted) || 
-	    (this != null) &&
-	    (casted != null) &&
-	    (getContext().nCtx() == casted.getContext().nCtx()) &&
-	    (Native.isEqFuncDecl(getContext().nCtx(), getNativeObject(), casted.getNativeObject()));
-    }
-
-    /**
-     * A hash code.
-     **/
-    public int hashCode()
-    {
-        return super.hashCode();
+        if (o == this) return true;
+        if (!(o instanceof FuncDecl)) return false;
+        FuncDecl other = (FuncDecl) o;
+
+        return
+            (getContext().nCtx() == other.getContext().nCtx()) &&
+            (Native.isEqFuncDecl(
+                getContext().nCtx(),
+                getNativeObject(),
+                other.getNativeObject()));
     }
 
-    /**
-     * A string representations of the function declaration.
-     **/
+    @Override
```

Comment:
`FuncDecl#toString` returns a string representation of the function declaration just based on the function name, doesn't it?